Post installation verifications for FaaS

jilju opened the pull request:
Creted new functions for FaaS provider and consumer verification.
Movied the verification of pods in managed-fusion namespace to the new function.
Verify security in ocs-client-operator pod on consumer.
Verifiy storage client on consumer.

Consumer:
    1. Verify CSV phase
    2. Verify client endpoint

Provider:
    1. Verify CSV phase
    2. Verify ocs-provider-server pod is Running
    4. Verify that Cephcluster is Ready and hostNetworking is True
    5. Verify that the security groups are set up correctly

yitzhak12 commented on ocs_ci/helpers/managed_services.py:
```diff
@@ -363,3 +370,235 @@ def verify_pods_in_managed_fusion_namespace():
+        sc_data = sc_obj.get()["items"][0]
+        verify_faas_provider_storagecluster(sc_data)
+        verify_faas_provider_resources()
+        if get_ocs_osd_deployer_version() >= get_semantic_version("2.0.11-0"):
```
This function `get_ocs_osd_deployer_version` should be updated for the `managed-fusion` namespace. (e.g., change the namespace to `config.ENV_DATA["cluster_namespace"]` and the constant `constants.OCS_OSD_DEPLOYER_CSV_LABEL` should be replaced.

yitzhak12 commented on ocs_ci/helpers/managed_services.py:
```diff
@@ -363,3 +370,235 @@ def verify_pods_in_managed_fusion_namespace():
+        verify_faas_provider_storagecluster(sc_data)
+        verify_faas_provider_resources()
+        if get_ocs_osd_deployer_version() >= get_semantic_version("2.0.11-0"):
+            verify_provider_topology()
```
I think this function should also be updated for the `managed-fusion` namespace.

jilju replied:
The function is already updated tin master to use the correct namespace.

yitzhak12 reviewed (commented):
The pr looks good

jilju commented:
Added other updates to unblock testing of the PR. 
1. Added checks for machineconfigpool before and after udating the pull secret.
2. Increased wit time for nodes after creating rosa cluster. The wait time is based on observations.

jilju commented:
Verified:
Provider - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/24229
Consumer - https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-deploy-ocs-cluster/24228
